pkg/controller/storagecluster/update.go

```diff
@@ -84,6 +85,37 @@ func (c *Controller) rollingUpdate(cluster *corev1.StorageCluster, hash string)
 	return c.syncNodes(cluster, oldPodsToDelete, []string{}, hash)
 }
 
+// annotateStoragePod annotate storage pods with custom annotations along with known annotations,
+// if no custom annotations created, only known annotations will be retained.
+func (c *Controller) annotateStoragePod(cluster *corev1.StorageCluster, pod *v1.Pod) error {
+	annotations := make(map[string]string)
+	// Keep only known storage pod annotations
+	if podAnnotations := pod.GetAnnotations(); podAnnotations != nil {
+		for _, knownKeys := range knownAnnotations {
+			if v, ok := podAnnotations[knownKeys]; ok {
+				annotations[knownKeys] = v
+			}
+		}
+	}
+	// Add custom annotations if exist
+	if customAnnotations := operatorutil.GetCustomAnnotations(cluster, corev1.PodAnnotations, corev1.Portworx); customAnnotations != nil {
+		for k, v := range customAnnotations {
+			annotations[k] = v
+		}
+	}
+	if len(annotations) == 0 {
+		pod.SetAnnotations(nil)
+	} else {
+		pod.SetAnnotations(annotations)
+	}
+	err := c.client.Update(context.TODO(), pod)
```

Comment:
I suggest check if custom annotations changed, if it is not changed, don't update the pod.

Reply:
with this change this function is more like a reconciler for custom annotations, then it makes more sense to put it in isPodUpdated.